src/ligero/tests.rs

```diff
@@ -345,4 +345,286 @@ mod tests {
         )
         .unwrap());
     }
+
+    #[test]
+    fn test_several_polynomials() {
+        let degrees = [4_usize, 13_usize, 30_usize];
+        let mut rng = &mut test_rng();
+
+        LigeroPCS::<2>::setup(*degrees.iter().max().unwrap(), None, rng).unwrap();
+        let leaf_hash_params = <LeafH as CRHScheme>::setup(&mut rng).unwrap();
+        let two_to_one_params = <CompressH as TwoToOneCRHScheme>::setup(&mut rng)
+            .unwrap()
+            .clone();
+
+        let ck: LigeroPCCommitterKey<MTConfig> = LigeroPCCommitterKey {
+            leaf_hash_params,
+            two_to_one_params,
+        };
+        let vk: LigeroPCVerifierKey<MTConfig> = LigeroPCVerifierKey {
+            leaf_hash_params,
+            two_to_one_params,
+        };
+
+        let rand_chacha = &mut ChaCha20Rng::from_rng(test_rng()).unwrap();
+        let mut test_sponge = test_sponge::<Fq>();
+        let mut challenge_generator: ChallengeGenerator<Fq, PoseidonSponge<Fq>> =
+            ChallengeGenerator::new_univariate(&mut test_sponge);
+
+        let mut labeled_polys = Vec::new();
+        let mut values = Vec::new();
+        
+        let point = Fq::rand(rand_chacha);
+
+        for degree in degrees {
+            let labeled_poly = LabeledPolynomial::new(
+                "test".to_string(),
+                rand_poly(degree, None, rand_chacha),
+                None,
+                None,
+            );
+
+            values.push(labeled_poly.evaluate(&point));
+            labeled_polys.push(labeled_poly);
+        }
+
+        let (commitments, randomness) = LigeroPCS::<2>::commit(&ck, &labeled_polys, None).unwrap();
+        
+        let proof = LigeroPCS::<2>::open(
+            &ck,
+            &labeled_polys,
+            &commitments,
+            &point,
+            &mut (challenge_generator.clone()),
+            &randomness,
+            None,
+        )
+        .unwrap();
+        assert!(LigeroPCS::<2>::check(
+            &vk,
+            &commitments,
+            &point,
+            values,
+            &proof,
+            &mut challenge_generator,
+            None
+        )
+        .unwrap());
+    }
+
+
+    #[test]
+    #[should_panic]
```

Comment:
You can also include a panic message, so that we're testing it panics at the right place

Reply:
Yes, you can write:
```
#[should_panic(expected = "stuff")]
```

Reply:
Cool feature! I Have now added expected panic messages to 3 out of the 5 tests which should panic. The exceptions are:
- `test_several_polynomials_swap_values`: In this case, the last check in `check` (namely, that the passed value is equal to the evaluation of the committed poly) fails and an error message is _printed_ informing about that, which does happen. However, the function does not panic - it returns Ok(false), since all that has happened is that the verification process has completed without errors. It is only the test function itself which panics, since it asserts it receives Ok(true). Perhaps we can rework the output of `check` (although the return format is imposed by the interface), but in any case that should not be addressed in this branch.

- `test_several_polynomials_swap_proofs`: In this case, one expects the proof verification process to go _horribly wrong somewhere_, since one is passing a proof for a different commitment! But it is hard a priori to know exactly at which point/how that proof verification will fail, and I don't know that the test should expect sth specific. Of course, I could just copy-paste the actual panic message which is displayed when the test is run, but I don't think that is the point.